src/components/atom/EntityQuantityInput.tsx

```diff
@@ -0,0 +1,32 @@
+import CrossIcon from '../icons/CloseIcon'
+import React from 'react'
+
+interface EntityQuantityInputProps {
+    quantity: number;
+    measurementUnit: string;
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    onClear: () => void;
+}
+const EntityQuantityInput = ({quantity, measurementUnit, onChange, onClear}: EntityQuantityInputProps) => {
+    return (
+        <div className="w-full h-24 px-8 bg-white border flex items-center gap-2 justify-between">
+            <div className="flex items-center w-full justify-center">
+                <input
+                    type="number"
+                    className="font-roboto-flex text-mega-input text-secondary-color w-28 text-right px-2"
+                    value={quantity || 0}
```

Comment:
since `quantity` must be provided, the only time `quantity` is "falsy" is when `quantity === 0` so I would just replace this with `{quantity}`